cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -220,15 +220,21 @@ static void sc_populate_libgl_with_hostfs_symlinks(const char *libgl_dir,
 	}
 }
 
-static void sc_mount_and_glob_files(const char *rootfs_dir,
-				    const char *tgt_dir,
-				    const char *glob_list[],
-				    size_t glob_list_len)
+static void sc_mount_and_mkdir_and_glob_files(const char *rootfs_dir,
+					      const char *tgt_dir,
+					      const char *glob_list[],
+					      size_t glob_list_len)
 {
 	// Bind mount a tmpfs on $rootfs_dir/$tgt_dir (i.e. /var/lib/snapd/lib/gl)
 	char buf[512] = { 0 };
 	sc_must_snprintf(buf, sizeof(buf), "%s%s", rootfs_dir, tgt_dir);
 	const char *libgl_dir = buf;
+
+	int res = mkdir(libgl_dir, 0755);
+	if (res != 0 && errno != EEXIST) {
+		die("cannot create tmpfs target %s", libgl_dir);
+	}
+
```

Comment:
tgt_dir is set to SC_LIBGL_DIR, which is /var/lib/snapd/lib/gl (or gl32 or vulkan). This PR only addresses the leaf directory and no parents. Is /var/lib/snapd/lib guaranteed to exist? What if SC_LIBGL_DIR (or its friends) changes to something like /var/lib/snapd/foo/bar/baz/lib/gl? If nothing else, there should be a comment at these `#define`s that makes this clear changing them will cause problems.